app/controllers/users/phones_controller.rb

```diff
@@ -22,13 +22,19 @@ def update
 
     private
 
+    # we only allow editing of the first configuration since we'll eventually be
+    # doing away with this controller. Once we move to multiple phones, we'll allow
+    # adding and deleting, but not editing.
+    def phone_configuration
+      current_user.phone_configurations.first
+    end
+
     def user_params
       params.require(:user_phone_form).permit(:phone, :international_code, :otp_delivery_preference)
     end
 
     def delivery_preference
-      current_user.phone_configurations.first&.delivery_preference ||
-        current_user.otp_delivery_preference
+      phone_configuration&.delivery_preference || current_user.otp_delivery_preference
```

Comment:
Because there are times when we update the otp delivery preference without having a phone configured, so we sometimes need to know the delivery preference before we have a phone. I don't know why we do that, but we do. I don't want to break anything.